### ISSUE TITLE
Error when trying to retrieve the temporal evolution of an entity

### ISSUE THREAD

**Issue by bobeal** [ScorpioBroker/ScorpioBroker#158]
Hello,

I created a temporal entity (via a POST request on `/ngsi-ld/v1/temporal/entities/`) with the following payload : 

```
{
	"id": "urn:ngsi-ld:Vehicle:A101",
	"type": "Vehicle",
	"speed": [
            {
		"type": "Property",
		"value": 80,
                "observedAt": "2020-11-06T17:00:00Z"
	    },
            {
		"type": "Property",
		"value": 90,
                "observedAt": "2020-11-06T18:00:00Z"
	    }
      ]
}
```

I received a 201 response with the expected `Location` header.

I then tried to do a temporal query with a GET on `/ngsi-ld/v1/temporal/entities/urn:ngsi-ld:Vehicle:A101` (and some variants with `attrs` parameter, time filters, ...) but only got empty responses.

So I looked into the logs and spotted the following error:

```
histmgr_1    | 2020-11-06T17:43:04,451 INFO  [http-nio-1040-exec-10] e.n.n.c.s.StorageReaderDAO: NGSI-LD to SQL: with r as (  select te.id, te.type, te.createdat, te.modifiedat, coalesce(teai.attributeid, '') as attributeid, jsonb_agg(teai.data  - 'https://uri.etsi.org/ngsi-ld/createdAt' - 'https://uri.etsi.org/ngsi-ld/modifiedAt' order by teai.modifiedat desc) as attributedata  from temporalentity te  left join temporalentityattrinstance teai on (teai.temporalentity_id = te.id)  where te.id='urn:ngsi-ld:Vehicle:A101' AND  1=1   group by te.id, te.type, te.createdat, te.modifiedat, teai.attributeid   order by te.id, teai.attributeid ) select id, tedata || case when attrdata <> '{"": [null]}'::jsonb then attrdata else tedata end as data from (   select id, ('{"@id":"' || id || '"}')::jsonb ||           ('{"@type":["' || type || '"]}')::jsonb   as tedata, jsonb_object_agg(attributeid, attributedata) as attrdata   from r   group by id, type, createdat, modifiedat   order by modifiedat desc ) as m
histmgr_1    | 2020-11-06T17:43:04,467 ERROR [http-nio-1040-exec-10] e.n.n.c.s.StorageReaderDAO: Exception ::
histmgr_1    | org.springframework.jdbc.IncorrectResultSetColumnCountException: Incorrect column count: expected 1, actual 2
histmgr_1    | 	at org.springframework.jdbc.core.SingleColumnRowMapper.mapRow(SingleColumnRowMapper.java:110)
histmgr_1    | 	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:94)
histmgr_1    | 	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:61)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:440)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.queryForList(JdbcTemplate.java:485)
histmgr_1    | 	at eu.neclab.ngsildbroker.commons.storage.StorageReaderDAO.query(StorageReaderDAO.java:44)
```

Is it a problem with my input data or is it on Scorpio's side?

Thanks in advance for your feedback!

Benoit.

**Comment by amitrnec** [comment]
Hi,
It is working. Below get request is working here. 
/ngsi-ld/v1/temporal/entities?type=Vehicle&attrs=speed&timerel=after&time=2010-08-14T12:00:00Z

Kindly  share what type of get request you are trying to do. 
Thanks.

**Comment by bobeal** [comment]
Hi,

I've just tried the same request than in your example and still got no response and the following error in the logs:

```
histmgr_1    | 2020-11-07T19:44:56,735 INFO  [http-nio-1040-exec-5] e.n.n.c.s.StorageReaderDAO: NGSI-LD to SQL: with r as (  select te.id, te.type, te.createdat, te.modifiedat, coalesce(teai.attributeid, '') as attributeid, jsonb_agg(teai.data  - 'https://uri.etsi.org/ngsi-ld/createdAt' - 'https://uri.etsi.org/ngsi-ld/modifiedAt' order by teai.modifiedat desc) as attributedata  from temporalentity te  left join temporalentityattrinstance teai on (teai.temporalentity_id = te.id)  where te.type='http://example.org/vehicle/Vehicle' AND teai.attributeid='http://example.org/vehicle/speed' AND ( (teai.static = true AND teai.observedat IS NULL) OR teai.observedat>= '2010-08-14T12:00:00Z'::timestamp) AND  1=1   group by te.id, te.type, te.createdat, te.modifiedat, teai.attributeid   order by te.id, teai.attributeid ) select id, tedata || case when attrdata <> '{"": [null]}'::jsonb then attrdata else tedata end as data from (   select id, ('{"@id":"' || id || '"}')::jsonb ||           ('{"@type":["' || type || '"]}')::jsonb   as tedata, jsonb_object_agg(attributeid, attributedata) as attrdata   from r   group by id, type, createdat, modifiedat   order by modifiedat desc ) as m
histmgr_1    | 2020-11-07T19:44:56,740 ERROR [http-nio-1040-exec-5] e.n.n.c.s.StorageReaderDAO: Exception ::
histmgr_1    | org.springframework.jdbc.IncorrectResultSetColumnCountException: Incorrect column count: expected 1, actual 2
histmgr_1    | 	at org.springframework.jdbc.core.SingleColumnRowMapper.mapRow(SingleColumnRowMapper.java:110)
histmgr_1    | 	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:94)
histmgr_1    | 	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:61)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:440)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
histmgr_1    | 	at org.springframework.jdbc.core.JdbcTemplate.queryForList(JdbcTemplate.java:485)
histmgr_1    | 	at eu.neclab.ngsildbroker.commons.storage.StorageReaderDAO.query(StorageReaderDAO.java:44)
```

Details of the HTTP request:

```
GET /ngsi-ld/v1/temporal/entities?type=Vehicle&attrs=speed&timerel=after&time=2010-08-14T12:00:00Z HTTP/1.1
Accept: application/ld+json
Link: <https://raw.githubusercontent.com/ScorpioBroker/ScorpioBroker/master/Examples/index.json>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"
User-Agent: PostmanRuntime/7.26.8
Cache-Control: no-cache
Host: 10.5.1.214:9090
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
```

PS: I'm running Scorpio in Docker mode, using `docker-compose-dist.yml` file from release FIWARE_7.9.

**Comment by amitrnec** [comment]
Hi,
I have tested with docker-compose-aaio.yml and scorpio:scorpio-aaio_latest .It is working with that. But there is some issue in  docker-compose-dist.yml file due to not updated in docker. However the issue has been fixed in Fiware 7.9 .

**Comment by bobeal** [comment]
Hi,

I've changed the Scorpio's Docker images tags in https://github.com/ScorpioBroker/ScorpioBroker/blob/FIWARE_7.9/docker-compose-dist.yml from 1.0.0 to FIWARE_7.9 and it works now!

Thanks!

Benoit.
